### DISCUSSION TITLE
Warning: The data area QSYS/QCPTOIMPF exists on this system and may impact Code for IBM i functionality

### DISCUSSION THREAD

**Discussion by zalmang** [codefori/vscode-ibmi#1815] [Q&A]
Anyone familiar with this error? 

Just wanted an idea of what's going on. What are these files for, why can't Code for i function with the older version, and what are the consequences of removing the older data area?

I didn't see anything listed on the Code for i forum.

Thank you!

<img width="344" alt="qcptoimpf" src="https://github.com/codefori/vscode-ibmi/assets/133536986/d040f583-a6af-48b7-964e-da559529d72c">

**Comment by worksofliam** [comment]
Hi @zalmang! 

What is it you want to know in addition to what the message is telling you?

Code for IBM I can't function as expected with that data in place.

**Reply by zalmang** [reply]
Just wanted an idea of what's going on. What are these files for, why can't Code for i function with the older version, and what are the consequences of removing it?

**Comment by smartrocks** [comment]
@Zalmang: As to the effect of the data areas read this: https://www.ibm.com/support/pages/cpyfrmimpf-and-cpytoimpf-data-areas 
You can probably remove them because is has to do with creation of PCASCII files using null or blank as a delimiter which you probably don't have. Apparently a small number of IBM customers were impacted when a new version of CPYTOIMPF/CPYFRMIMPF was introduced, and the data areas were provided as a work around to use the old version.
